load.php

```diff
@@ -8,26 +8,60 @@
 	die( 'Not an entry point.' );
 }
 
-SemanticMediaWiki::initExtension();
-
-$GLOBALS['wgExtensionFunctions'][] = function() {
-	SemanticMediaWiki::onExtensionFunction();
-};
+/**
+ * ExtensionRegistry only maps classes and functions after all extensions have
+ * been queued from the LocalSettings.php resulting in DefaultSettings not being
+ * loaded in-time.
+ *
+ * When changing the load order, please ensure that this function is run either
+ * via Composer's autoloading or as part of your internal registration.
+ */
+SemanticMediaWiki::load();
 
 /**
  * @codeCoverageIgnore
  */
 class SemanticMediaWiki {
 
 	/**
-	 * @since 2.4
+	 * @since 2.5
+	 *
+	 * @note It is expected that this function is loaded before LocalSettings.php
+	 * to ensure that settings and global functions are available by the time
+	 * the extension is activated.
 	 */
-	public static function initExtension() {
+	public static function load() {
 
 		if ( is_readable( __DIR__ . '/vendor/autoload.php' ) ) {
 			include_once __DIR__ . '/vendor/autoload.php';
 		}
 
+		include_once __DIR__ . '/src/Aliases.php';
+		include_once __DIR__ . '/src/Defines.php';
+		include_once __DIR__ . '/src/GlobalFunctions.php';
+
+		$settings = include __DIR__ . '/DefaultSettings.php';
+
+		foreach ( $settings as $key => $value ) {
+			$GLOBALS[$key] = $value;
+		}
+
+		unset( $settings );
```

Comment:
> the include into a getDefaultSettings function, though I think the later is overkill here

Wikia/app#11922 changed the settings file to get access to individual settings hence the anticipation of this by providing that in a form without someone having to edit DefaultSettings.

They have to make some changes eventually but I'd rather see them to do those changes in this file with less potential of conflict.